Compass/ParseServer.swift

```diff
@@ -25,55 +31,116 @@ class PServer {
        // PFUser.enableAutomaticUser()
         let defaultACL = PFACL()
         defaultACL.hasPublicReadAccess = true
-        
         PFACL.setDefault(defaultACL, withAccessForCurrentUser: true)
-        
     }
     
-    func saveUserLocation(classN: String, uData: UserData) {
+    func logIn(userName: String, pass: String, completition: @escaping (PFUser?, Error?) -> Void) {
+        PFUser.logInWithUsername(inBackground: userName, password: pass) { (user, error) in
+            guard error == nil else {
+                completition (nil, error)
+                return
+            }
+            if user != nil {
```

Comment:
I'd probably make it so that the completion is called even if user is nil, since whatever calls this may well rely on the completion block being called.
This does mean you'd have to handle the case when there's no error from `PFUser.logInWithUsername` and no user though, which I'd probably do by creating my own error to return